src/neo/Network/P2P/TaskManager.cs

```diff
@@ -271,6 +271,12 @@ private void RequestTasks(TaskSession session)
             else if (Blockchain.Singleton.HeaderHeight >= session.LastBlockIndex
                     && TimeProvider.Current.UtcNow.ToTimestamp() - PingCoolingOffPeriod >= Blockchain.Singleton.GetBlock(Blockchain.Singleton.CurrentHeaderHash)?.Timestamp)
             {
+                if (session.ShouldAskForMemPool)
+                {
+                    session.ShouldAskForMemPool = false;
```

Comment:
Does this assignment of `false` needs to happen every time RequestTasks is called?
It looks good to me, @shargon.
However, we need to take care that we are not going to spam the P2P as I mentioned. It is not related to this specific PR, however, it looks like we need protection, otherwise nodes could try to keep asking for mempool when it is full during possible spam attacks.

Reply:
It should happen only once

Reply:
I understood now. AHEuahea I got confused with the `session.ShouldAskForMemPool` value.